[Bug] [Connector-Jdbc] Writing to MySQL does not support collection data types such as: set type.

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

Writing to MySQL does not support collection data types such as: set type

### SeaTunnel Version

2.3.12

### SeaTunnel Config

```conf
default config
```

### Running Command

```shell
--config ./config/v2.batch.config.mysql -m local
```

### Error Exception

```log
Caused by: org.apache.seatunnel.common.utils.SeaTunnelException: org.apache.seatunnel.common.exception.SeaTunnelRuntimeException: ErrorCode:[COMMON-20], ErrorDescription:['MySQL' table 'chestnufang_batch_test_sink.test_0' unsupported get catalog table with field data types '{"hobbies":"SET"}']
	at org.apache.seatunnel.common.exception.CommonError.getCatalogTableWithUnsupportedType(CommonError.java:177)
	at org.apache.seatunnel.connectors.seatunnel.jdbc.catalog.AbstractJdbcCatalog.buildColumnsWithErrorCheck(AbstractJdbcCatalog.java:267)
	at org.apache.seatunnel.connectors.seatunnel.jdbc.catalog.AbstractJdbcCatalog.buildColumnsReturnTablaSchemaBuilder(AbstractJdbcCatalog.java:245)
	at org.apache.seatunnel.connectors.seatunnel.jdbc.catalog.AbstractJdbcCatalog.getTable(AbstractJdbcCatalog.java:218)
	at org.apache.seatunnel.connectors.seatunnel.jdbc.sink.JdbcSink.getDatabaseTableSchema(JdbcSink.java:195)
	at org.apache.seatunnel.connectors.seatunnel.jdbc.sink.JdbcSink.createWriter(JdbcSink.java:150)
	at org.apache.seatunnel.connectors.seatunnel.jdbc.sink.JdbcSink.createWriter(JdbcSink.java:66)
	at org.apache.seatunnel.api.sink.multitablesink.MultiTableSink.createWriter(MultiTableSink.java:82)
```

### Zeta or Flink or Spark Version

zeta 

### Java or Scala Version

java 1.8

### Screenshots

<img width="1714" height="417" alt="Image" src="https://github.com/user-attachments/assets/588827a1-4a6f-4206-8a09-e2ddfcb4fd5e" />

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/mysql/MySqlTypeConverter.java
+++ b/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/mysql/MySqlTypeConverter.java
@@ -75,6 +75,7 @@ public class MySqlTypeConverter implements TypeConverter<BasicTypeDefine<MysqlTy
     static final String MYSQL_LONGTEXT = "LONGTEXT";
     static final String MYSQL_JSON = "JSON";
     static final String MYSQL_ENUM = "ENUM";
+    static final String MYSQL_SET = "SET";
 
     // ------------------------------time-------------------------
     static final String MYSQL_DATE = "DATE";
@@ -243,6 +244,7 @@ public class MySqlTypeConverter implements TypeConverter<BasicTypeDefine<MysqlTy
                 builder.scale(decimalUnsignedType.getScale());
                 break;
             case MYSQL_ENUM:
+            case MYSQL_SET:
                 builder.dataType(BasicType.STRING_TYPE);
                 if (typeDefine.getLength() == null || typeDefine.getLength() <= 0) {
                     builder.columnLength(100L);

--- a/seatunnel-connectors-v2/connector-jdbc/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/mysql/MySqlTypeConverterTest.java
+++ b/seatunnel-connectors-v2/connector-jdbc/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/mysql/MySqlTypeConverterTest.java
@@ -1082,4 +1082,20 @@ public class MySqlTypeConverterTest {
         Assertions.assertEquals(MySqlTypeConverter.MYSQL_DATETIME, typeDefine.getColumnType());
         Assertions.assertEquals(MySqlTypeConverter.MYSQL_DATETIME, typeDefine.getDataType());
     }
+
+    @Test
+    public void testConvertSet() {
+        BasicTypeDefine<Object> typeDefine =
+                BasicTypeDefine.builder()
+                        .name("test")
+                        .columnType("SET('reading','sports','music','travel')")
+                        .dataType("SET")
+                        .length(3L)
+                        .build();
+        Column column = MySqlTypeConverter.DEFAULT_INSTANCE.convert(typeDefine);
+        Assertions.assertEquals(typeDefine.getName(), column.getName());
+        Assertions.assertEquals(BasicType.STRING_TYPE, column.getDataType());
+        Assertions.assertEquals(3, column.getColumnLength());
+        Assertions.assertEquals(typeDefine.getColumnType(), column.getSourceType());
+    }
 }

